examples/gno.land/r/gnoland/coins/coins.gno

```diff
@@ -0,0 +1,171 @@
+// - "<denom>/<address>": Show specific address balance for the given coin
+func Render(path string) string {
+	if path == "" {
+		return renderHomepage()
+	}
+
+	if !strings.Contains(path, "/") {
+		denom := path
+		totalSupply := TotalSupply(denom)
+		return renderCoinInfo(denom, totalSupply)
+	}
+
+	parts := strings.Split(path, "/")
+	if len(parts) != 2 {
+		panic(errInvalidPath)
+	}
+
+	denom, addr := parts[0], parts[1]
+	return renderAddressBalance(denom, std.Address(addr))
+
+	// TODO: implement an explanatory homepage that shows usage instructions
+	// TODO: implement a coin info page showing total supply, and other metrics if we have them
+	// TODO: implement an address page showing balance for specified coin
+	// TODO: create navigation links between pages to enable intuitive exploration without typing URLs
+}
+
+func renderHomepage() string {
+	var sb strings.Builder
+
+	sb.WriteString(md.H1("Gno.land Coins"))
+	sb.WriteString(md.Paragraph("This realm provides an interface to query coin information on the Gno.land blockchain."))
+
+	sb.WriteString(md.H2("Usage"))
+	sb.WriteString(md.Paragraph("You can query information using the following URL patterns:"))
+
+	urlPatterns := []string{
+		md.Link("/r/gnoland/coins:ugnot", "/r/gnoland/coins:ugnot") + " - Check total supply of ugnot",
+		"/r/gnoland/coins:ugnot/{address} - Check ugnot balance of a specific address",
+	}
+	sb.WriteString(md.BulletList(urlPatterns))
+
+	sb.WriteString(md.H2("Supported Coins"))
+	supportedCoins := []string{
+		md.Link("ugnot", "/r/gnoland/coins:ugnot") + " - The native token of Gno.land",
+	}
+	sb.WriteString(md.BulletList(supportedCoins))
+
+	return sb.String()
+}
+
+func renderCoinInfo(denom string, totalSupply int64) string {
+	var sb strings.Builder
+
+	sb.WriteString(md.H1("Coin Information"))
+
+	sb.WriteString(md.H2("Denomination"))
+	sb.WriteString(md.Paragraph(denom))
+
+	sb.WriteString(md.H2("Total Supply"))
+	sb.WriteString(md.Paragraph(std.NewCoin(denom, totalSupply).String()))
+
+	sb.WriteString(md.H2("Statistics"))
+	stats := []string{
+		"Total Supply: " + std.NewCoin(denom, totalSupply).String(),
+		"Chain ID: " + std.ChainID(),
+		"Current Block Height: " + strconv.Itoa(int(std.ChainHeight())),
+	}
+	sb.WriteString(md.BulletList(stats))
+
+	sb.WriteString(md.Paragraph(md.Link("Back to Home", "/r/gnoland/coins")))
+
+	return sb.String()
+}
+
+func renderAddressBalance(denom string, address std.Address) string {
+	var sb strings.Builder
+	balance := AddressBalance(denom, address)
+	isEligible := IsEligible(denom, address)
+
+	sb.WriteString(md.H1("Address Balance Information"))
+
+	sb.WriteString(md.H2("Address"))
+	sb.WriteString(md.Paragraph(address.String()))
+
+	sb.WriteString(md.H2("Balance"))
+	sb.WriteString(md.Paragraph(std.NewCoin(denom, balance).String()))
+
+	sb.WriteString(md.H2("Additional Information"))
+	info := []string{
+		"Airdrop Eligibility: " + (map[bool]string{true: "Eligible", false: "Not Eligible"})[isEligible],
```

Comment:
do NOT use the term airdrop anywhere in this realm; it's a balance checker, not an airdrop checker.

Reply:
removed in here
https://github.com/gnolang/gno/pull/3899/commits/5c175b20ac9c94c068e2e31f7a0da4bc1afcda08